AmericanMedicalAssociation/joe-style-guide
# EWLJ-874: Update article tooltips for keyboard navigation and screen …

…reader use.



## Ticket(s)

Please do not submit a Pull Request without a relevant JIRA ticket or Github issue! If you are creating a new pattern or feature, create an issue describing the need for this feature in Github **first** and then link to it below.

**Github Issue**

- [Issue XXX: Issue Title](https://github.com/AmericanMedicalAssociation/joe-style-guide/issues/XXX)

**Jira Ticket**

- [EWL-874: JOE Article Tooltip (Issue 66246)](https://ama-it.atlassian.net/browse/EWLJ-874)

- [EWL-878: JOE Article Footnotes (Issue 66800)](https://ama-it.atlassian.net/browse/EWLJ-878)


## Description

Note this PR covers the functionality for both tickets listed above. Updated the script and styling handling JOE tooltips to enable tooltips using keyboard navigation. Updated script to allow tooltip sup numbers and their corresponding tooltips to be read aloud by screen readers.


## To Test

For Keyboard Navigation:
- Link styleguide changes locally
- clear caches
- On any article page begin tabbing through article content 
- Confirm on tab through that any and footnote reference links (i.e <sup>#</sup> elements) are highlighted
- Confirm when a footnote reference link is focused/tabbed on the tooltip appears
- On continuation of tab through, confirm any links within that tooltip are next in the sequence
- Confirm on continuation again, the next link or footnote reference in the article is focused and the previous tooltip disappears

For screen reader use:
- **NOTE**: The PR was tested using google chromes screen reader extension [Chrome Screen Reader](https://chromewebstore.google.com/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn?pli=1). Other screen readers may be used, results may vary.
- Link styleguide changes locally
- clear caches
- With screen reader select a paragraph with a footnote reference link
- Wait and listen to automated output, and confirm when a footnote reference link is read, the aria-label is read, for example for "Footnote Link 1" should be read aloud
- Next, click or tab to a footnote reference link confirm after the aria label is read aloud like the content within the tooltip is read aloud after
- On tab through of a toolip confirm any links within that tooltip are also read aloud 

## Visual Regressions

Does your work introduce any visual regressions to existing work in the Style Guide, or in a Drupal 8 environment? Please either call these out here or address them before marking your PR as `ready for review`.


## Relevant Screenshots/GIFs
![Screenshot 2024-12-06 at 12 46 27 PM](https://github.com/user-attachments/assets/c3affdc2-fb06-4ca8-b58a-3c29b53bff02)

![Screenshot 2024-12-06 at 12 46 35 PM](https://github.com/user-attachments/assets/b4397e71-d77c-494e-9f9e-bd63f48d2a41)

![Screenshot 2024-12-06 at 12 47 17 PM](https://github.com/user-attachments/assets/d072665e-c2c0-4ab1-bc63-703815037aa7)


## Remaining Tasks

Remaining tasks?


## Additional Notes

Anything more to add? Good things to call out here are:
- are there any PRs in this or other repositories that relate to or affect this PR?
- are there any caveats or oddities testers should know about when testing this PR?
- are there any problems you ran into during your work that you'd like to call out?
